test/cli_test.py

```diff
@@ -340,35 +342,44 @@ def mock_get_pty_info(shell: bool) -> api_pb2.PTYInfo:
             env_term_program=os.environ.get("TERM_PROGRAM"),
         )
 
+    captured_out = []
+
+    async def write_to_fd(fd: int, data: bytes):
+        nonlocal captured_out
+        captured_out.append((fd, data))
+
     with mock.patch("rich.console.Console.is_terminal", True), mock.patch(
         "modal._pty.get_pty_info", mock_get_pty_info
-    ), mock.patch("modal._pty.write_stdin_to_pty_stream", asyncnullcontext):
-        yield
+    ), mock.patch("modal._container_exec.get_pty_info", mock_get_pty_info), mock.patch(
+        "modal._pty.write_stdin_to_pty_stream", asyncnullcontext
+    ), mock.patch(
+        "modal._container_exec.handle_exec_input", asyncnullcontext
+    ), mock.patch(
+        "modal._container_exec._write_to_fd", write_to_fd
```

Comment:
Could we use pytest's [capsys](https://docs.pytest.org/en/7.1.x/how-to/capture-stdout-stderr.html#accessing-captured-output-from-a-test-function) fixture instead of mocking `write_to_fd`?

Reply:
I think I tried it and it didn't work, but I can test this again -- maybe I made a mistake while testing.

Reply:
Oh, maybe I need `capfd` instead of `capsys`.

Reply:
Never mind, I still can't get it to work, even with `capfd` and `capfdbinary`. Maybe `loop.add_writer(fd, try_write)` doesn't work (ie. `try_write` never gets called)?